crates/bifrost/src/loglet_wrapper.rs

```diff
@@ -42,6 +42,7 @@ pub struct LogletWrapper {
     pub(crate) base_lsn: Lsn,
     /// If set, it points to the first first LSN outside the boundary of this loglet (bifrost's tail semantics)
     pub(crate) tail_lsn: Option<Lsn>,
+    pub(crate) config: LogletConfig,
```

Comment:
LogletWrapper now has a copy of LogletConfig. This will make it easier to inspect the writable segment config without checking metadata which is prone to race-conditions.

Only drawback is that we will have to copy the config from the Segment on creation.

Reply:
I don't think I'm too concerned about the memory usage (perhaps wrongly so) but I'm curious about the race conditions you see. Is it because the logs config metadata might get independently updated? Given that it's mutable, it seems ok to me to retain a copy of the config we obtained on creation of the wrapper.